src/mito2/src/compaction.rs

```diff
@@ -414,6 +434,24 @@ impl PendingCompaction {
     }
 }
 
+/// Finds TTL of table by first examine table options then database options.
+async fn find_ttl(
+    table_id: TableId,
+    table_ttl: Option<Duration>,
+    schema_metadata_manager: &SchemaMetadataManagerRef,
+) -> Result<Option<Duration>> {
+    if let Some(table_ttl) = table_ttl {
+        return Ok(Some(table_ttl));
+    }
+
+    let ttl = schema_metadata_manager
+        .get_schema_options_by_table_id(table_id)
```

Comment:
Consider a timeout setting to prevent the engine from being blocked by slow network I/O.